lib/public/models/DataExportModel.js

```diff
@@ -0,0 +1,207 @@
+     * @param {boolean} disabled if true disabled, otherwise enabled
+     */
+    setDisabled(disabled) {
+        this._disabled = disabled;
+    }
+
+    /**
+     * If the model has no data to be exported, it calls for it via action function provided in constructor
+     * @return {Promise<void>|null} promise if data were not asked for, null otherwise
+     */
+    callForData() {
+        if (this._onDataNotAskedAction) {
+            return this.items.match({
+                NotAsked: () => this._onDataNotAskedAction(),
+                Other: () => null,
+            });
+        } else {
+            throw new Error('onDataNotAskedAction was not provided');
+        }
+    }
+
+    /**
+     * Return the current items remote data
+     *
+     * @return {RemoteData<T[]>} the items
+     */
+    get items() {
+        return this._items$.getCurrent();
+    }
+
+    /**
+     * Observable notified when the export configuration visually changes
+     *
+     * @return {Observable} the visual change observable
+     */
+    get visualChange$() {
+        return this._visualChange$;
+    }
+
+    /**
+     * Get export type selected by the user
+     *
+     * @return {string} export type
+     */
+    get selectedExportType() {
+        return this._selectedExportType;
+    }
+
+    /**
+     * Set export type
+     *
+     * @param {string} exportType export type
+     * @return {void}
+     */
+    setSelectedExportType(exportType) {
+        this._selectedExportType = exportType;
+        this.notify();
+        this._visualChange$.notify();
+    }
+
+    /**
+     * Get selected fields
+     *
+     * @return {string[]} selected fields
+     */
+    get selectedFields() {
+        return this._selectedFields;
+    }
+
+    /**
+     * Update selected fields from HTML options list
+     *
+     * @param {HTMLCollection|Array} selectedOptions options collection
+     * @return {void}
+     */
+    setSelectedFields(selectedOptions) {
+        this._selectedFields = [];
+        [...selectedOptions].forEach(({ value }) => this._selectedFields.push(value));
+        this.notify();
+        this._visualChange$.notify();
+    }
+
+    /**
+     * Create export using current items observable
+     *
+     * @return {Promise<void>} void
+     */
+    async createExport() {
+        this.items.apply({
+            Success: (items) => {
+                const { selectedFields } = this;
+
+                const formatted = items.map((item) => {
+                    const selectedEntries = Object.entries(pick(item, selectedFields));
+                    const mappedEntries = selectedEntries.map(([key, value]) => {
+                        const formatter = this.dataExportConfiguration[key]?.exportFormat || ((v) => v);
+                        return [key, formatter(value, item)];
+                    });
+
+                    return Object.fromEntries(mappedEntries);
+                });
+
+                this.selectedExportType === 'CSV'
```

Comment:
The default value you use in the constructor is `JSON`. Would it be best to also assume JSON first here to be in-line.